src/builder.ts

```diff
@@ -96,22 +105,25 @@ export default class Builder {
 						self.layers.push(sha)
 					}
 
-					this.emit('data', data.stream)
+					if (data.stream) {
+						outputStream.write(data.stream);
+					}
```

Comment:
* See sample values for the data object a few lines above. `data.stream` is sometimes `undefined`, hence the test. 
* The previous code here was the last line in the following block:
```
const outputStream = res
.pipe(JSONStream.parse())
.pipe(es.through(function(data: any): void {
    ...
    this.emit('data', data.stream)
```
The last line, does it mean that the event stream was emitting a new `'data'` event into itself? I found it unclear whether this would trigger the `'data'` handler again. Anyway, I found that the output event-stream could be replaced with node's [PassThrough](https://nodejs.org/docs/latest-v6.x/api/stream.html#stream_class_stream_passthrough) stream, for the same reasons mentioned above for the `inputStream`. And I found it clearer to run `outputStream.write(data.stream)`, as this avoids a stream writing to itself.